### PR TITLE
Fix: Glossary for Cloud

### LINKED ISSUE #35519: Add glossary: cloud
### MDN URL

https://developer.mozilla.org/en-US/docs/glossary

### What specific section or headline is this issue about?

glossary

### What information was incorrect, unhelpful, or incomplete?

We mention "cloud" and "cloud computing" but don't really define it. WOuld be good to include a brief definiton.

### What did you expect to see?

cloud included in the MDN glossary.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by ragul1697**
### Description
Fixes #35519 - Added Glossary item for `Cloud`

### Additional details
Since MDN focuses on the web, it might be beneficial to include content about hosting web applications. Additionally, adding information about cloud services and cloud service providers could be valuable.

### CODE REVIEW

File: files/en-us/glossary/cloud/index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: Cloud
+slug: Glossary/Cloud
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+**Cloud** or **Cloud Computing** refers to the on-demand delivery of a wide range of computing services such as servers, storage, databases, networking, software, analytics, and intelligence via the Internet.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/glossary/cloud/index.md:9]: You could also consider mentioning data centers, and provide some examples of some of the services you mention above.

I'd remove "software" from your list. Most if not all of them are software, so it seems to be a bit too vague.

[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/glossary/cloud/index.md:9]: Also, when you say "intelligence", what do you mean? AI? Again, I'd remove that, unless you want to instead provide examples of tools that use AI or LLMs.

File: files/en-us/glossary/cloud/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+This model enables faster innovation, flexible scalability, and significant cost savings. Users can access and utilize these services through a _pay-as-you-go_ pricing model, ensuring they only pay for what they use, making it a highly efficient and adaptable solution for businesses and individuals alike.
+
+## See also
+
+- [What is Cloud Computing?](https://cloud.google.com/learn/what-is-cloud-computing?hl=en) on Google Cloud
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud/index.md:15]: Also link to https://en.wikipedia.org/wiki/Cloud_computing?

File: files/en-us/glossary/cloud/index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+title: Cloud
+slug: Glossary/Cloud
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+**Cloud** or **Cloud Computing** refers to the on-demand delivery of a wide range of computing services such as servers, storage, databases, networking, software, analytics, and intelligence via the Internet.
+
+This model enables faster innovation, flexible scalability, and significant cost savings. Users can access and utilize these services through a _pay-as-you-go_ pricing model, ensuring they only pay for what they use, making it a highly efficient and adaptable solution for businesses and individuals alike.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud/index.md:11]: Mention "Software as a service" here, and its relationship with cloud pay-as-you-go and subscription models?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Hi there @ragul1697, and thank you for your contribution to MDN! I've provided you with a few comments to consider, although I think this is largely going in the right direction.

**Comment by ragul1697** [comment]
Thanks for all the comments. I will make the changes soon and let you know.

**Comment by cwangsanata** [comment]
I think another good addition would be to include different cloud deployment style (e.g. Private Cloud, Public Cloud, etc.) as mentioned in the [NIST Definition of Cloud Computing](https://nvlpubs.nist.gov/nistpubs/legacy/sp/nistspecialpublication800-145.pdf). Also including the document itself could be a good source for additional reading.

**Comment by ragul1697** [comment]
Hi @chrisdavidmills, I have made some changes for the Glossary item on Cloud Computing. Pease have a look at it. Thanks

File: files/en-us/glossary/cloud/index.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+title: Cloud
+slug: Glossary/Cloud
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+**Cloud** refers to servers that are accessed over the Internet, and the software and databases that run on those servers.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud/index.md:9]:
```suggestion
The **Cloud** refers to servers located worldwide that are accessed over the Internet, and the software and databases that run on those servers.
```

File: files/en-us/glossary/cloud/index.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+Instead of storing data or running programs on a local computer or server, users rely on these remote resources. Cloud servers are located in data centers all over the world. By using {{glossary("Cloud Computing")}}, users and companies do not have to manage physical servers themselves or run software applications on their own machines.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud/index.md:10]:
```suggestion
Instead of storing data or running programs on a local computer or server, users can access cloud resources to perform tasks: this is known as {{glossary("Cloud computing")}}.
```

File: files/en-us/glossary/cloud/index.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+### Key features of the Cloud
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud/index.md:12]:
```suggestion
### Key features of the cloud
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
+---
+title: Cloud Computing
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:2]:
```suggestion
title: Cloud computing
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+slug: Glossary/Cloud_Computing
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:3]:
```suggestion
slug: Glossary/Cloud_computing
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+**Cloud Computing** refers to the on-demand delivery of a wide range of computing services such as servers, storage, databases, networking, analytics, and intelligence via the Internet.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:9]:
```suggestion
**Cloud computing** refers to the on-demand delivery of a wide range of computing services such as servers, storage, databases, networking, analytics, and intelligence via the Internet — also known as **cloud services**.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+This model enables faster innovation, flexible scalability, and significant cost savings. Users can access and utilize these services through a _pay-as-you-go_ pricing model, ensuring they only pay for what they use, making it a highly efficient and adaptable solution for businesses and individuals alike.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:11]:
```suggestion
This model enables faster innovation, flexible scalability, and significant cost savings. Users can use cloud-based services through a _pay-as-you-go_ pricing model, ensuring they only pay for what they use, making them highly efficient and adaptable solutions for businesses and individuals alike.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+> [!NOTE]
+> The {{glossary("Cloud")}} in Cloud Computing refers to the server infrastructure on which cloud services are built and hosted.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:14]:
```suggestion
> The {{glossary("Cloud")}} in Cloud computing refers to the server infrastructure on which cloud services are built and hosted.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+### Cloud Services
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:16]: I'd suggest just deleting this section. I've added in the cloud services term to the opening paragraph in an earlier section, so this now seems repetitious. Also, the second sentence here isn't needed. That's pretty obvious to the target audience.

File: files/en-us/glossary/cloud_computing/index.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+Cloud services are infrastructure, platforms, or software that are hosted by third-party providers and made available to users through the internet. Users can access cloud services with nothing more than a computer, operating system, and internet connectivity.
+
+### Cloud Service Providers (CSP)
+
+Cloud Service Providers are the companies which provides these cloud services. Most popular cloud service providers are Google (Google Cloud Platform), Amazon AWS (Amazon Web Services), and Microsoft (Azure). They offer similar services with different price models, features and so on.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:22]:
```suggestion
**Cloud service providers** are companies that provide cloud services, such as Google (Google Cloud Platform), Amazon AWS (Amazon Web Services), and Microsoft (Azure).
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+### Types of Cloud Computing
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:24]:
```suggestion
### Types of cloud computing
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+The three main types of cloud computing include Infrastructure as a Service(IaaS), Platform as a Service(PaaS), and Software as a Service(SaaS). Each type of cloud computing provides different levels of control, flexibility, and management so that you can select the right set of services for your needs.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:26]:
```suggestion
The three main types of cloud computing are Infrastructure as a Service (IaaS), Platform as a Service (PaaS), and Software as a Service (SaaS). Each type of cloud computing provides different levels of control, flexibility, and management.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+#### Infrastructure as a Service
+
+IaaS provides virtualized computing resources over the internet, including servers, storage, and networking, allowing users to rent and manage the infrastructure they need which gives the users highest level of flexibility and control over the resources.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:30]:
```suggestion
IaaS provides virtualized computing resources over the internet, including servers, storage, and networking, allowing users to manage their infrastructure. This gives users a high level of flexibility and control over the resources.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+
+> **Use case:** A company needing to scale its server infrastructure rapidly without owning physical servers might use IaaS.
+
+**Examples:** Amazon Web Services (AWS EC2), Microsoft Azure Virtual Machines, Google Compute Engine (GCE)
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:34]:
```suggestion
- **Examples:** Amazon Web Services (AWS EC2), Microsoft Azure Virtual Machines, Google Compute Engine (GCE).
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+
+### Types of Cloud Computing
+
+The three main types of cloud computing include Infrastructure as a Service(IaaS), Platform as a Service(PaaS), and Software as a Service(SaaS). Each type of cloud computing provides different levels of control, flexibility, and management so that you can select the right set of services for your needs.
+
+#### Infrastructure as a Service
+
+IaaS provides virtualized computing resources over the internet, including servers, storage, and networking, allowing users to rent and manage the infrastructure they need which gives the users highest level of flexibility and control over the resources.
+
+> **Use case:** A company needing to scale its server infrastructure rapidly without owning physical servers might use IaaS.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:32]:
```suggestion
- **Use case:** A company needing to scale its server infrastructure rapidly without owning physical servers.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:33]: Delete the space in between. Copy this pattern for the other sections too.

File: files/en-us/glossary/cloud_computing/index.md [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+**Examples:** Amazon Web Services (AWS EC2), Microsoft Azure Virtual Machines, Google Compute Engine (GCE)
+
+#### Platform as a Service
+
+PaaS provides a platform allowing customers to develop, run, and manage applications without worrying about the underlying infrastructure (like servers, storage, and networking).
+
+> **Use case:** A developer who wants to focus on writing code and application logic, without managing servers or operating systems, would use PaaS.
+
+Examples: Google App Engine, Microsoft Azure App Service, Heroku, AWS Elastic Beanstalk
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:42]:
```suggestion
- **Examples**: Google App Engine, Microsoft Azure App Service, Heroku, AWS Elastic Beanstalk.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+
+> **Use case:** A company needing to scale its server infrastructure rapidly without owning physical servers might use IaaS.
+
+**Examples:** Amazon Web Services (AWS EC2), Microsoft Azure Virtual Machines, Google Compute Engine (GCE)
+
+#### Platform as a Service
+
+PaaS provides a platform allowing customers to develop, run, and manage applications without worrying about the underlying infrastructure (like servers, storage, and networking).
+
+> **Use case:** A developer who wants to focus on writing code and application logic, without managing servers or operating systems, would use PaaS.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:40]:
```suggestion
- **Use case:** A developer who wants to focus on writing code and application logic, without managing servers or operating systems.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+
+Examples: Google App Engine, Microsoft Azure App Service, Heroku, AWS Elastic Beanstalk
+
+#### Software as a Service
+
+SaaS delivers software applications over the internet, which users can access via a web browser. The underlying infrastructure, platform, and data are all managed by the provider.
+
+> **Use case:** A business using SaaS doesn't need to install or manage software locally. Instead, they access the software (like email, CRM, or collaboration tools) directly from a web browser.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:48]:
```suggestion
- **Use case:** A business using SaaS doesn't need to install or manage software locally. Instead, they access the software (like email, CRM, or collaboration tools) directly from a web browser.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+
+Examples: Google Workspace, Microsoft 365, Slack, Github, ChatGPT
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:50]:
```suggestion
- **Examples**: Google Workspace, Microsoft 365, Slack, Github, ChatGPT.
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+
+### Cloud Deployment MOdels
```
Comment (chrisdavidmills) [inline files/en-us/glossary/cloud_computing/index.md:52]: I would delete this whole section. I think the above section covering Iaas/PaaS/SaaS is useful, as it gives readers an idea of different types of Cloud service. But going into different cloud deployment models goes beyond providing a useful definition of cloud computing ad isn't needed in this article.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@ragul1697 great work., thank you so much. This content is interesting and useful. I've provided a bunch of comments, but most of them are fairly minor grammar/consistency updates.

The only really major one is the last one; I really don't think you need the last section.

File: files/en-us/glossary/cloud_computing/index.md (hunk 24 above)
Comment (ragul1697) [reply files/en-us/glossary/cloud_computing/index.md:52]: Sure, I will remove it. Can we add the [NIST Definition of Cloud Computing](https://nvlpubs.nist.gov/nistpubs/legacy/sp/nistspecialpublication800-145.pdf) under see also section? I think that document is some kind of standard but the contents are explained in other references also.

**Comment by ragul1697** [comment]
Hi @chrisdavidmills, completed all the changes requested. Thank you✌️

File: files/en-us/glossary/cloud/index.md [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+The **Cloud** refers to servers located worldwide that are accessed over the Internet, and the software and databases that run on those servers.
+
+Instead of storing data or running programs on a local computer or server, users can access cloud resources to perform tasks: this is known as {{glossary("Cloud computing")}}.
+
+### Key features of the cloud
```
Comment (Josh-Cena) [inline files/en-us/glossary/cloud/index.md:13]:
``` suggestion
Key features of the cloud include:
```

File: files/en-us/glossary/cloud_computing/index.md [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+page-type: glossary-definition
+---
+
+{{GlossarySidebar}}
+
+**Cloud computing** refers to the on-demand delivery of a wide range of computing services such as servers, storage, databases, networking, analytics, and intelligence via the Internet — also known as **cloud services**.
+
+This model enables faster innovation, flexible scalability, and significant cost savings. Users can use cloud-based services through a _pay-as-you-go_ pricing model, ensuring they only pay for what they use, making them highly efficient and adaptable solutions for businesses and individuals alike.
+
+### Cloud Services
```
Comment (Josh-Cena) [inline files/en-us/glossary/cloud_computing/index.md:13]: Try to remove as many of these headings as possible. Glossaries are short and don't need many subsections. Also pay attention that you must start from h2 and go down, not start from h3.

Comment (ragul1697) [reply files/en-us/glossary/cloud_computing/index.md:13]: Yes, I understand. Glossary should be small and I started this entry like that. But I had to add more content to it as requested by others. At the moment MDN doesn't have any resources related to Cloud and Cloud Computing. 

In future, it might be useful if MDN provides guides for Cloud topics related to Web dev as most of the current technologies are using Cloud servers for various purposes.

**Review by Josh-Cena — APPROVED** [review]
I've taken liberty to reorganize the "cloud computing" page so it is much terser and doesn't use headings, but all information is still there. This looks great and we just need approval from @chrisdavidmills

**Review by chrisdavidmills — APPROVED** [review]
OK, approving! 

Thanks to @Josh-Cena for helping to tidy this up. I also made a small edit to tighten the text up a bit further (the second paragraph was a bit wordy, and used "cloud services" too often). I am really happy with this now.

Our biggest thanks here must go to @ragul1697 for sticking with us through all the change requests and producing a nice piece of content for MDN. Your contribution is much appreciated!

**Comment by ragul1697** [comment]
Thank you @chrisdavidmills.
